client/components/RegisterPage.tsx

```diff
@@ -1,64 +1,133 @@
 import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { 
+  TextField, 
+  Box, 
+  Typography, 
+  InputAdornment,
+  IconButton,
+  Button
+} from "@mui/material";
+import {
+  Visibility,
+  VisibilityOff
+} from "@mui/icons-material";
 import { useRegisterMutation } from "../features/api";
 
 const RegisterPage = () => {
   const [register, { isLoading, isError, data}] = useRegisterMutation();
 
+  const navigate = useNavigate();
+
   const [email, setEmail] = useState("");
   const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
   const [confirmPassword, setConfirmPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const handleClickShowConfirmPassword = () => setShowConfirmPassword((show) => !show);
+  const handleMouseDownConfirmPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };  
 
   const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    register({ email, username, password });
+    if (password === confirmPassword) {
+      register({ email, username, password });
+      navigate("/")
+    } else {
+      alert("Password confirmation does not match");
```

Comment:
This should be a React error that is displayed on the UI - if you want to come back for it later when you add more detailed form validation, please leave a TODO

Reply:
Added TODO

Reply:
Note: if you have a GitHub issue created for this already, you can add it to the TODO, eg `TODO #32 Change this to a React error that is displayed in the UI`